### ISSUE TITLE
[package] libkml/1.3.0: libkml missing minizip/crypt.h

### ISSUE THREAD

**Issue by Renari** [conan-io/conan-center-index#6986]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **libkml/1.3.0**
  * Operating System+version: **Linux Ubuntu 19.10**
  * Compiler+version: **GCC 9**
  * Docker image: **conanio/gcc9**
  * Conan version: **conan 1.39.0**
  * Python version: **Python 3.7.5**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=9
compiler.libcxx=libstdc++
build_type=Release
[options]
[conf]
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)
```
conan install libkml/1.3.0@_/_ --build=libkml --build=missing
```
It appears that libkml is sourcing it's own version of minizip that does not include the `crypt.h` file that it is including. 

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
/home/conan/.conan/data/libkml/1.3.0/_/_/build/c2f82833e15443d3910adede4610191beda2324b/source_subfolder/src/kml/base/contrib/minizip/unzip.c:183:10: fatal error: minizip/crypt.h: No such file or directory
  183 | #include <minizip/crypt.h>
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [source_subfolder/src/kml/base/CMakeFiles/kmlbase.dir/build.make:108: source_subfolder/src/kml/base/CMakeFiles/kmlbase.dir/contrib/minizip/unzip.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:272: source_subfolder/src/kml/base/CMakeFiles/kmlbase.dir/all] Error 2
make: *** [Makefile:149: all] Error 2
libkml/1.3.0: 
libkml/1.3.0: ERROR: Package 'c2f82833e15443d3910adede4610191beda2324b' build failed
libkml/1.3.0: WARN: Build folder /home/conan/.conan/data/libkml/1.3.0/_/_/build/c2f82833e15443d3910adede4610191beda2324b
ERROR: libkml/1.3.0: Error in build() method, line 54
	cmake.build()
	ConanException: Error 2 while executing cmake --build '/home/conan/.conan/data/libkml/1.3.0/_/_/build/c2f82833e15443d3910adede4610191beda2324b/build_subfolder' '--' '-j16'

```
</details>

**Comment by SpaceIm** [comment]
Yes, we were relying on minizip coming from zlib recipe. I'll fix that.

**Comment by SpaceIm** [comment]
Oh, I know what's wrong. It comes from a modification of minizip recipe following this discussion: https://github.com/conan-io/conan-center-index/pull/6641#issuecomment-891280522

**Comment by sh0** [comment]
@SpaceIm So the problem with leaving `minizip/crypt.h` in minizip package in my case was with building Python, exactly same as #1525.

Looking at libkml it seems that they basically bundle version 1.01e of minizip from [here](https://www.winimage.com/zLibDll/minizip.html) under `src/kml/base/contrib/minizip` with some changes added on later. For some unclear reason they opted to not bundle the `crypt.h` and to do `#include <minizip/crypt.h>` instead.

Some options:

* Bundle `crypt.h` with libkml.
* Add `crypt.h` back to minizip, remove `os.path.join("include", "minizip")` from `self.cpp_info.includedirs` in minizip and patch any software that uses minizip to use `#include <minizip/*.h>` style includes.
* Add `crypt.h` back to minizip and start patching all other software that uses glibc `crypt.h` and minizip in same package. That is actually pretty hard to do as you would likely need to still get rid of the minizip directory include in build files and then still add `#include <minizip/*.h>` style includes.

**Comment by SpaceIm** [comment]
> Bundle crypt.h with libkml.

Probably not an option in libkml recipe.

> Add crypt.h back to minizip, remove os.path.join("include", "minizip") from self.cpp_info.includedirs in minizip and patch any software that uses minizip to use #include <minizip/*.h> style includes.

Not an option, unfortunately it's not the official include dir of minizip.

> Add crypt.h back to minizip and start patching all other software that uses glibc crypt.h and minizip in same package. That is actually pretty hard to do as you would likely need to still get rid of the minizip directory include in build files and then still add #include <minizip/*.h> style includes.

Probably the best option, crypt.h is an official header of minizip. I don't know whether author is aware of the conflict.

**Comment by sh0** [comment]
There is a PR from 2017 still open [here](https://github.com/madler/zlib/pull/229). FYI Debian and Ubuntu keep `crypt.h`. Gentoo, Arch Linux, Fedora remove `crypt.h` from minizip.

**Comment by sh0** [comment]
Fourth option: Add [this](https://github.com/libkml/libkml/commit/a03adc24bc8cf5ab76a0e65f2756172a72c28b7a) patch from upstream to libkml to disable including `crypt.h` when using bundled minizip. Did not have time to test it though.

Arch Linux has same problem with libkml. See discussion [here](https://aur.archlinux.org/packages/libkml/).

**Comment by SpaceIm** [comment]
Yes we can try this libkml PR.

**Comment by SpaceIm** [comment]
I've added the fix in an existing PR https://github.com/conan-io/conan-center-index/pull/5684
